docs/source/en/tasks/object_detection.mdx

```diff
@@ -0,0 +1,487 @@
+...     num_train_epochs=10,
+...     fp16=False,
+...     save_steps=200,
+...     logging_steps=50,
+...     learning_rate=1e-5,
+...     save_total_limit=2,
+...     remove_unused_columns=False,
+...     push_to_hub=True,
+... )
+```
+
+Finally, bring everything together, and call `train`:
+
+```py
+>>> from transformers import Trainer
+
+>>> trainer = Trainer(
+...     model=model,
+...     args=training_args,
+...     data_collator=collate_fn,
+...     train_dataset=cppe5["train"],
+...     tokenizer=image_processor,
+... )
+
+>>> rainer.train()
+```
+
+## Evaluate
+Object detection models are commonly evaluated with a set of [COCO-style metrics](https://cocodataset.org/#detection-eval).
+You can use one of the existing metrics implementations, here you'll use the one from `torchvision`.
+
+Save your trained model to use for evaluation. Alternatively, you can evaluate the model that you have pushed to the Hub.
+
+```py
+>>> trainer.save_model("./detr-resnet-50_fine_tuned_cppe5")
+```
+
+To use the `torchvision` evaluator, you'll need to prepare a ground truth COCO dataset. The API to build a COCO dataset
+requires the data to be stored in a certain format, so you'll need to save images and annotations to disk first. Just like
+when you prepared your data for training, the annotations from the `cppe5["test"]` need to be formatted. However, images
+should stay as they are.
+
+The following example shows how you can:
+- Create a torchvision CocoDetection dataset
+- Reformat annotations from the `cppe5["test"]`
+- Save images and annotations into files that you can create an instance of CocoDetection dataset from
+- Use `cocoevaluate` metric to compute the [COCO metrics](https://cocodataset.org/#detection-eval).
+
+```py
+import os
+import json
+import evaluate
+import torchvision
+import numpy as np
+from tqdm import tqdm
+
+
+class CocoDetection(torchvision.datasets.CocoDetection):
+    def __init__(self, img_folder, feature_extractor, ann_file):
+        super(CocoDetection, self).__init__(img_folder, ann_file)
+        self.feature_extractor = feature_extractor
+
+    def __getitem__(self, idx):
+        # read in PIL image and target in COCO format
+        img, target = super(CocoDetection, self).__getitem__(idx)
+
+        # preprocess image and target (converting target to DETR format, resizing + normalization of both image and target)
+        image_id = self.ids[idx]
+        target = {"image_id": image_id, "annotations": target}
+        encoding = self.feature_extractor(images=img, annotations=target, return_tensors="pt")
+        pixel_values = encoding["pixel_values"].squeeze()  # remove batch dimension
+        target = encoding["labels"][0]  # remove batch dimension
+
+        return {"pixel_values": pixel_values, "labels": target}
+
+
+im_processor = DetrImageProcessor.from_pretrained("./detr-resnet-50_fine_tuned_cppe5")
+model = DetrForObjectDetection.from_pretrained("./detr-resnet-50_fine_tuned_cppe5")
```

Comment:
We usually load the fine-tuned models from the Hub. But I don't have strong opinions about it.

Reply:
I will probably switch to loading from the Hub. Then I won't need to save locally either, and it'll be slightly more consistent. My initial thought was that I might like to evaluate first before sharing. But it doesn't change that much here.

Reply:
Fixed in 1a38260b27ef977e0d82ba1c918eeb19c8f3dcb7